wandb/cli/cli.py

```diff
@@ -959,6 +959,12 @@ def launch_sweep(
     scheduler_args: Dict[str, Any] = parsed_user_config.pop("scheduler", {})
     settings: Dict[str, Any] = scheduler_args.pop("settings", {})
 
+    # update scheduler resource args from launch args
+    if launch_args.get("resource") and not scheduler_args.get("resource"):
+        scheduler_args["resource"] = launch_args["resource"]
+    if launch_args.get("resource_args") and not scheduler_args.get("resource_args"):
+        scheduler_args["resource_args"] = launch_args["resource_args"]
```

Comment:
maybe: 
```suggestion
        scheduler_args["resource_args"] = scheduler_args.get("resource_args", {}).update(launch_args["resource_args"])
```

Reply:
is this actually desirable though?  It feels like we're violating principle of least surprise... could one reasonably deduce that in the general case, launch resource args are automatically overlaid over scheduler args?  I think there's a faulty assumption baked in here, but I'm not too familiar with the motivating use case so I might be wrong.  cc @jlzhao27 any preference here?

Reply:
Right, so: 
```suggestion
        scheduler_args["resource_args"] = launch_args["resource_args"].update(scheduler_args.get("resource_args", {})
```

Reply:
Thats clearly better

Reply:
hmm this has the same issue.  the faulty assumption i'm referring to is that the `resource_args` meant for the launched job generally can and should apply to the scheduler job's resource. in particular, consider the case where the scheduler is running as a local process, and the launched jobs are running as k8s.  Contrived examples:

- a scheduler job running as a local process, but the actual launched sweep runs are k8s. 
- a custom scheduler job that needs one class of GPU, launching sweep runs that need a different class of GPU

Reply:
Yeah I guess in this case you would have to explicitly specify the resource for the scheduler too. Because these are overrides, i see your point that they would perhaps be more likely be used to target one or the other, rather than both. 
Let me take another pass at it.

Reply:
The more i think about it the more I agree with you, they should be entirely separate. Not that I can't think of many examples of where the above is useful, but managing the inheritance adds considerable complexity.